scala-package/dev/compile-mxnet-backend.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# (Yizhi) This is mainly inspired by the script in apache/spark.
+# I did some modificaiton to get it with our project.
+#
+
+set -e
+echo "Compiling MXNet Backend, Hang tight!....."
+
+if [[ ($# -ne 3) || ( $1 == "--help") ||  $1 == "-h" ]]; then
+  echo "Usage: [-h|--help]  <osx-x86_64-cpu/linux-x86_64-cpu/linux-x86_64-gpu> <project.basedir>" 1>&2
+  exit 1
+fi
+PLATFORM=$1
+MXNETDIR=$2
+
+
+# below routine shamelessly copied from
+# https://github.com/apache/incubator-mxnet/blob/master/setup-utils/install-mxnet-osx-python.sh
+
+chkret() {
+	cmd=$*
+	echo "$cmd"
+	$cmd
+	ret=$?
+	if [[ ${ret} != 0 ]]; then
+		echo " "
+		echo "ERROR: Return value non-zero for: $cmd"
+		echo " "
+		exit 1
+	fi
+} # chkret()
+
+UNAME=`uname -s`
+chkret pushd $MXNETDIR
+chkret git submodule update --init --recursive
+
+# don't want to overwrite an existing config file
+cp make/config.mk ./config.mk
+
+if [[ $PLATFORM == "osx-x86_64-cpu" ]];
+then
+    echo "Building MXNet Backend on MAC OS"
+    echo "ADD_CFLAGS += -I/usr/local/opt/opencv@2/include" >> ./config.mk
```

Comment:
Need to fix the opencv@2 here. Currently MXNet Python package use different openCV version

Reply:
Yes, lets fix this in the next iteration during release. I don't know at the moment what will work.

Reply:
Here is the issue to track https://github.com/apache/incubator-mxnet/issues/1119

Reply:
During release?